File: python/gpt/p20.py
# p20: read value from command line and add a constant offset of 1000
import sys

OFFSET = 1000


def main():
    if len(sys.argv) < 2:
        print("Usage: python p20.py <number>")
        return 1
    try:
        x = int(sys.argv[1])
    except Exception:
        print("Invalid number")
        return 1
    print(x + OFFSET)
    return 0


if __name__ == "__main__":
    main()

File: javascript/gpt/p20.js
// p20: read value from command line and add constant OFFSET = 1000
const OFFSET = 1000;
if (process.argv.length < 3) {
  console.log("Usage: node p20.js <number>");
  process.exit(1);
}
const x = parseInt(process.argv[2], 10);
if (Number.isNaN(x)) {
  console.log("Invalid number");
  process.exit(1);
}
console.log(x + OFFSET);
